src/remote.cc

```diff
@@ -682,6 +688,102 @@ static void gr_slideshow_delay(const gchar *text, GIOChannel *, gpointer)
 	options->slideshow.delay = static_cast<gint>(n * 10.0 + 0.01);
 }
 
+static void gr_duplicates_threshold(const gchar *text, GIOChannel *, gpointer)
+{
+	gint thresh;
+	gint n;
+	gint res;
+
+	res = sscanf(text, "%d", &thresh);
+	if (res == 1)
+		{
+		n = thresh;
+		if (n < 0 || n > 100)
+			{
+			printf_term(TRUE, "Image similarity threshold out of range (%d to %d)\n", 0, 100);
+			return;
+			}
+		}
+	else
+		{
+		n = 99;
+		}
+
+	options->duplicates_similarity_threshold = static_cast<guint>(n);
+	DEBUG_0("threshold set to %d", options->duplicates_similarity_threshold);
+}
+
+static void gr_duplicates_program(const gchar *text, GIOChannel *, gpointer)
+{
+	g_strdup(options->duplicates_program);
+	options->duplicates_program = g_strdup(text);
+	DEBUG_0("duplicates program set to \"%s\"", options->duplicates_program);
+}
+
+static void gr_process_duplicates(const gchar *, GIOChannel *, gpointer data)
+{
+	auto remote_data = static_cast<RemoteData *>(data);
+    std::map<std::string, std::unique_ptr<pic>> pics;
+	GList *work = remote_data->file_list;
+	while (work)
+	{
+		FileData *fd = static_cast<FileData *>(work->data);
+        std::string name(fd->path);
+        pics[name] = std::unique_ptr<pic>(new pic(fd->path));
+		work = work->next;
+	}
+	DEBUG_1("processing %d files in set", pics.size());
+
+	// Compute similarity score for every pair, build equivalence sets.
+    for (auto a = pics.begin(); a != pics.end(); ++a) {
```

Comment:
Please use meaningful variable names for temporaries.  Maybe `left_iter` and `right_iter`?  `left_pic` and `right_pic`?  Doesn't have to be those, but the meanings should be self-evident from the name.